### ISSUE TITLE
web3.eth.filter watch C++ logs zeros but Go logs data

### ISSUE THREAD

**Issue by ethers** [ethereum/cpp-ethereum#1369]
cpp develop branch commit 7b6710a

Have this dapp code, using ethereum.js develop branch commit de5de9e
```
web3.eth.filter({'address': gExchangeContractAddr}).watch(function(res) {
  console.log('@@@ gExchangeContractAddr filter')
  console.log(res)
});
```

With Go cli commit 4f5b362, the console logs as expected (data of 1):
```
@@@ gExchangeContractAddr filter
Object {address: "0x12393873e2d12de2f9ec7bf51803356e49eb922f", topic: Array[1], data: "0x0000000000000000000000000000000000000000000000000000000000000001", number: 16268}
```

cpp eth cli just logs zeros:
Object {address: "0x0000000000000000000000000000000000000000", data: "0x", hash: "0x0000000000000000000000000000000000000000000000000000000000000000", number: 0, topics: Array[0]}

**Comment by gavofyork** [comment]
ethereum.js calls `eth_newFilter` with options object including `"toBlock"` defined as `0x0` by default. this is wrong - it should be either left undefined or defined as `"latest"`.

@debris look into this would'ya :)

**Comment by gavofyork** [comment]
Fixed in e57d6ae..07c5cc3

**Comment by ethers** [comment]
I built cpp develop eth cli 0ff3080 (pull, cmake, make) and ethereum.js develop de5de9e (npm run-script build and copied to my dapp)

With this code:
```
web3.eth.filter({'address': gExchangeContractAddr}).watch(function(res) {
  console.log('@@@ gExchangeContractAddr filter')
  console.log(res)
});
```
I only see the filter fire one time: it is at the very page load:
```
@@@ gExchangeContractAddr filter
null
```

After the sendTransaction() call, I expect it to get fired and log some data.  (rpc query shows that the contract code went through a flow where a log() was done [which is what Go logs]).  [I have also checked that my value for gExchangeContractAddr is correct.]


I also tried adding toBlock
`web3.eth.filter({'address': gExchangeContractAddr, 'toBlock': 'latest'}).watch....` and it led to #1368

**Comment by ethers** [comment]
ethereum.js fixed 9f073d9 - applied fixes from cpp-ethereum
